### DISCUSSION TITLE
Symmetric encrypted data is not decrypted in this framework or in any other OpenPGP tools

### DISCUSSION THREAD

**Discussion by skumarm** [krzyzanowskim/ObjectivePGP#209] [General]
- ObjectivePGP version: 0.15.0
- Platform (iOS/macOS): iOS
- Installation method (CocoaPods/Carthage/...): CocoaPods
- Project name: Learning

# Description

I used the same method as described in https://github.com/krzyzanowskim/ObjectivePGP/issues/114#issuecomment-395199469. 

But I am getting the following error...

Error Domain=com.objectivepgp Code=9 "Unable to decrypt. Invalid message or missing key to decrypt message." UserInfo={NSLocalizedDescription=Unable to decrypt. Invalid message or missing key to decrypt message.}

Please help me to fix this...

# Steps to reproduce

1. First encrypt a string 
2. Then decrypt the PGPArmor data for the encrypted data

Sample code to reproduce the issue:

```objective-c

// Code goes here

NSError *encryptError;
NSData *encData = [ObjectivePGP encrypt:[@"Santhosh Kumar M" dataUsingEncoding:NSUTF8StringEncoding] addSignature:NO usingKeys:@[] passphraseForKey:^NSString * _Nullable(PGPKey * _Nonnull key) {
            return @"dooth";
        } error:&encryptError];

NSString *encString = [PGPArmor armored:encData as:PGPArmorMessage];

NSError *decryptError;
NSData *decData = [ObjectivePGP decrypt:[encString dataUsingEncoding:NSUTF8StringEncoding] andVerifySignature:YES usingKeys:@[] passphraseForKey:^NSString * _Nullable(PGPKey * _Nullable key) {
            return @"dooth";
        } error:&decryptError];

```

**Comment by skumarm** [comment]
Getting the following errors while decrypting by the **gpg** command line tool.

gpg: assuming IDEA encrypted data
gpg: [don't know]: invalid packet (ctb=4f)
gpg: mdc_packet with invalid encoding
gpg: decryption failed: Invalid packet
gpg: onepass_sig with unknown version 29

**Comment by Appelg** [comment]
I have the same problem, did you find a solution to this?

Edit: 
It actually seems to be a problem with "encryption" function. The key function is never called / hit if I set a breakpoint. So it seem to produce invalid output.
The encrypted data cannot be decrypted on Linux either.

But if I encrypt data on Linux with GPG, it's possible to decrypt it with ObjectivePGP.
